### PR TITLE
Move some pages to development section

### PR DESCRIPTION
**Pull request by sagikazarmark**
This PR moves some development pages to a new section called development.

Some remaining tasks:

- [x] There is no actual content on the development page (yet? maybe there should be?), so it shouldn't show up in the side menu as a link
- [x] Development should show up as the last section in the menu
- [x] Contributing should be moved here as soon as #69 is merged

@nate-double-u can you help me out here? I'm not a huge Hugo expert as you are. ;)

### CODE REVIEW

**Comment by nate-double-u** [comment]
The location of menu items on the sidebar is based on the `weight` value in each page's frontmatter. I've revised all the frontmatter `weight` values to make it clearer what the groups are, and where each page should sit in the menu. (Sorry for inflating the files changed count!).

I also fixed an issue where "On this page:" was showing up even if the `toc` frontmatter variable was set to `false`.

**Comment by nate-double-u** [comment]
> There is no actual content on the development page (yet? maybe there should be?), so it shouldn't show up in the side menu as a link

This may be a bigger issue. We are using the [section-nav.html](https://github.com/dexidp/website/blob/main/layouts/partials/section-nav.html) file to build this component, and it's not doing any checks for content in the pages that it's using to build the menu.

I've only just started using Hugo myself, but now I'm curious if there is a way of checking for empty files -- if we can check, then we can render that menu content a little differently. My usual workaround has been to add content to those top-level menu pages 😅️

**Comment by sagikazarmark** [comment]
> This may be a bigger issue. We are using the section-nav.html file to build this component, and it's not doing any checks for content in the pages that it's using to build the menu.

It doesn't matter that much. We can write some random text about "This section explains how you can....blahblah".

**Comment by nate-double-u** [comment]
> We can write some random text about "This section explains how you can....blahblah".

This is definitely the easier solution 😉️

**Comment by nate-double-u** [comment]
I'll keep thinking on this menu issue though, this is going to come up more and more as the Hugo Netlify Template gets used. If we implement a solution there I'll look at merging it downstream.

File: content/docs/development/dev-dependencies.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ description: ""
 date: 2020-09-30
 draft: false
 toc: true
-weight: 100
+weight: 3100
```
Comment (nate-double-u) [inline content/docs/development/dev-dependencies.md:7]: I think that the default sort is alphabetical when files have the same weight.
`dev-integration-tests.md` and `dev-releases.md` both have a weight of 3100 set. I'd suggest being more explicit about the weights, even if you want alphabetical sorting, I'd set the weights that way. 
(lower weight gets higher precedence)

**Comment by sagikazarmark** [comment]
I think this is ready to be merged. Further improvements can be done in additional PRs.

File: content/docs/development/_index.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+title: "Development"
```
Comment (nate-double-u) [inline content/docs/development/_index.md:2]: At some point we may want to put a list of the development pages on this page. We could probably write a partial to build the list for us automatically.

File: content/docs/development/_index.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+description: ""
+date: 2020-12-09
+draft: false
+toc: false
+weight: 3000
+---
+
+This section explains how you can develop Dex and documents what steps needs to be done at each phase of development (setting up the development environment, running tests, submitting changes, releasing software, etc).
```
Comment (nate-double-u) [inline content/docs/development/_index.md:10]:
```suggestion
This section explains how you can develop for the Dex project and documents what steps need to be done at each phase of development (setting up the development environment, running tests, submitting changes, releasing software, etc).
```

**Review by nate-double-u — APPROVED** [review]
One minor nit, otherwise LGTM

File: content/docs/development/_index.md (hunk 2 above)
Comment (sagikazarmark) [reply content/docs/development/_index.md:2]: Sounds good to me.

**Comment by sagikazarmark** [comment]
Thanks for the review @nate-double-u

**Pull request merged**